### PR TITLE
feat(output): implement output strategies

### PR DESCRIPTION
**Pull request by GregoryAlbouy**

### CODE REVIEW

File: output/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+//go:build !prod
```
Comment (moreirathomas) [inline output/endpoint.go:1]: How does this react if we simply run `go run main.go` ?

File: requester/report.go [hunk 2]
```diff
@@ -21,38 +20,32 @@ func (rep Report) String() string {
...
-
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return fmt.Errorf("%w: %s", ErrReporting, err)
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode >= 400 {
-		return fmt.Errorf("%w: %s", ErrReporting, resp.Status)
+// makeReport generates and returns a Report from a previous Run.
+func makeReport(records []Record, numErr int, d time.Duration) Report {
```
Comment (moreirathomas) [inline requester/report.go:43]: As it only instanciate a new report without doing anything else: 
```suggestion
func newReport(records []Record, numErr int, d time.Duration) Report {
```
?

File: output/export/export.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	// ErrHTTPResponse reports an HTTP response error, such as bad status code.
+	ErrHTTPResponse = errors.New("export: server response error")
+)
+
+// Interface gathers the necessary methods to use any function exposed
+// in this package.
+type Interface interface {
+	fmt.Stringer
+	HTTPRequester
+}
```
Comment (moreirathomas) [inline output/export/export.go:29]: Smart !

File: output/strategy.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		}
+		if o == config.OutputJSON {
+			s |= JSONFile
+		}
+		if o == config.OutputBenchttp {
+			s |= Benchttp
+		}
+	}
+	return s
+}
```
Comment (moreirathomas) [inline output/strategy.go:39]: Very nice masking, give elegant fallthrough behaviour on use !

**Review by moreirathomas — APPROVED** [review]
Great jo, nicely decoupled too ! 👍 

Just comments on some naming.

File: output/endpoint.go (hunk 1 above)
Comment (GregoryAlbouy) [reply output/endpoint.go:1]: This `go:build` directive tells the compiler to include the file only if the build tag `prod` is not set, as a consequence the behavior of `go run ./cmd/runner` does not change. To summarize:

```sh
> go run ./cmd/runner
# endpoint is http://localhost:9998/report

> go run -tags prod ./cmd:runner
# endpoint is https://api.benchttp.app/report
```

File: config/file/parse.go [hunk 5]
```diff
@@ -45,12 +74,18 @@ func Parse(cfgpath string) (cfg config.Global, err error) {
 // parseRawConfig parses an input raw config as a config.Config and returns it
 // or the first non-nil error occurring in the process.
 func parseRawConfig(raw unmarshaledConfig) (config.Global, error) { //nolint:gocognit // acceptable complexity for a parsing func
+	const numField = 10 // should match the number of config Fields (not critical)
+
 	cfg := config.Global{}
-	fields := make([]string, 0, 9)
+	fields := make([]string, 0, numField)
+
+	appendField := func(field string) {
```
[the remark this answers is not in the document]
Comment (GregoryAlbouy) [reply config/file/parse.go:82]: Given `appendField` is just an alias to `fields = append(fields, field)` I find its name accurate, or am I missing something?

[the remark this answers is not in the document]
Comment (moreirathomas) [reply config/file/parse.go:82]: Nevermind ! ^^

**Review by moreirathomas — APPROVED** [review]

**Pull request merged**